odl/solvers/functional/functional.py

```diff
@@ -1462,15 +1432,15 @@ def initialized(self):
     def _initialize_if_needed(self):
         """Helper function to initialize the functional."""
         if not self.__is_initialized:
-            self.__func_subgrad_eval = self.func_subgrad(self.point)
+
             self.__constant = (-self.functional(self.point) +
-                               self.__func_subgrad_eval.inner(self.point))
+                               self.__subgrad.inner(self.point))
```

Comment:
Absolutely correct. The `__` attributes should be used during `__init__` at least until the class is finalized (`super()` has been called). After that the "official" attributes are preferred since that allows subclasses to customize their behavior.